drivers/serial/uart_ns16550.c

```diff
@@ -613,7 +668,18 @@ static int uart_ns16550_irq_tx_ready(struct device *dev)
  */
 static int uart_ns16550_irq_tx_complete(struct device *dev)
 {
-	return (INBYTE(LSR(dev)) & (LSR_TEMT | LSR_THRE)) == (LSR_TEMT | LSR_THRE);
+	k_spinlock_key_t key = k_spin_lock(&DEV_DATA(dev)->lock);
+	int check = 0;
+
+	if (
```

Comment:
Personal opinion: this way of breaking up lines looks horrible, particularly the way the closing paren before the equality check is on a separate line from the opening paren, but also the bare opening paren following `if`.

Maybe this could be done instead, satisfying both the line length limit and MISRA's "ints aren't bools" rule.

```
int rv = ((INBYTE(LSR(dev)) & (LSR_TEMT | LSR_THRE))
          == (LSR_TEMT | LSR_THRE)) ? 1 : 0;
```

Several other functions below also could use a trinary expression to avoid using an `if` to override an initialized variable.

Not a required change.

Reply:
Wow, that helps clean it up a lot. Thanks!